litestar/utils/signature.py

```diff
@@ -253,16 +254,46 @@ def add_types_to_signature_namespace(
     Args:
         signature_types: A list of types to add to the signature namespace.
         signature_namespace: The signature namespace to add types to.
+        logger: a logging facility to optional use for warning
 
     Raises:
-        ImproperlyConfiguredException: If a type is already defined in the signature namespace.
         AttributeError: If a type does not have a `__name__` attribute.
 
     Returns:
         The updated signature namespace.
     """
-    for typ in signature_types:
-        if (name := typ.__name__) in signature_namespace:
-            raise ImproperlyConfiguredException(f"Type '{name}' is already defined in the signature namespace")
-        signature_namespace[name] = typ
+    return merge_signature_namespaces(
+        signature_namespace=signature_namespace,
+        additional_signature_namespace={signature_type.__name__: signature_type for signature_type in signature_types},
+        logger=logger,
+    )
+
+
+def merge_signature_namespaces(
```

Comment:
Should use this in `BaseRouteHandler.resolve_signature_namespace` as well for consistency :)